File: content-solutions/graph-representation/graph-two-transfers.java
class Solution {
    public int[][] solution(int[][] routes) {
        int n = routes.length;
        int[][] answer = new int[n][n];
        for (int i = 0; i < n; i++) {
            for (int k = 0; k < n; k++) {
                if (routes[i][k] == 0) continue;
                for (int j = 0; j < n; j++) {
                    if (routes[k][j] == 1) answer[i][j] = 1;
                }
            }
        }
        return answer;
    }
}

File: content-solutions/graph-representation/graph-two-transfers.js
function solution(routes) {
  const n = routes.length;
  const answer = routes.map(() => new Array(n).fill(0));
  for (let i = 0; i < n; i++) {
    for (let k = 0; k < n; k++) {
      if (!routes[i][k]) continue;
      for (let j = 0; j < n; j++) {
        if (routes[k][j]) answer[i][j] = 1;
      }
    }
  }
  return answer;
}
